tools/update_sample_data.py

```diff
@@ -0,0 +1,102 @@
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#   "numpy",
+#   "astropy",
+#   "sunpy[net]",
+#   "dkist",
+# ]
+# ///
+"""
+This script recreates the sample data files and uploads the recreated versions to Asgard.
+"""
+import sys
+import tarfile
+import argparse
+from pathlib import Path
+
+import numpy as np
+
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+
+import dkist
+import dkist.net
+from dkist.net.globus import start_transfer_from_file_list, watch_transfer_progress
+from dkist.net.globus.endpoints import get_local_endpoint_id, get_transfer_client
+
+datasets = {
+    "AJQWW": {
+        "tiled": True,
+        "tile_slice": np.s_[0],
+        "filename": "AJQWW_single_mosaic.tar",
+    },
+    "BKPLX": {
+        "tiled": False,
+        "slice": np.s_[0],
+        "filename": "BKPLX_stokesI.tar",
+    },
+}
+
+def main(datasets, working_directory, destination_path="/user_tools_tutorial_data/"):
+    working_directory = Path(working_directory)
+    working_directory.mkdir(parents=True, exist_ok=True)
+    sample_files_for_upload = []
+
+    for did, props in datasets.items():
+        res = Fido.search(a.dkist.Dataset(did))
+        asdf_file = Fido.fetch(res, path=working_directory / "{dataset_id}", progress=False, overwrite=False)
+
+        ds = dkist.load_dataset(asdf_file)
+        if "slice" in props:
+            ds = ds[props["slice"]]
```

Comment:
Don't we need to save the sliced dataset metadata file? I.e., right now the metadata file provided as part of the sample data would refer to more than the sample data files, right?

Reply:
That's correct, but that's ok, the idea is that some files are missing which is a supported usecase.